framework/auth/views.py

```diff
@@ -163,6 +164,27 @@ def auth_logout(redirect_url=None):
     return resp
 
 
+def auth_email_logout(token, user):
+    """When a user is adding an email or merging an account, add the email to the user and log them out.
+    """
+    redirect_url = web_url_for('auth_login') + '?existing_user=True'
+    if user.confirm_token(token):
+        try:
+            user_merge = User.find_one(Q('emails', 'iexact', user.email_verifications[token]['email']))
```

Comment:
Don't use `iexact` unless absolutely necessary -- it's very slow. 
`Q('emails', 'eq', user.email_verifications[token]['email'].lower())` should work